tests/rbd/test_rbd_deep_flatten_negative.py

```diff
@@ -0,0 +1,111 @@
+
+from ceph.rbd.initial_config import initial_rbd_config
+from ceph.rbd.utils import getdict
+from ceph.rbd.workflows.cleanup import cleanup
+from ceph.rbd.workflows.rbd import create_snap_and_clone
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def test_deep_flatten_negative_scenario(rbd_obj, **kw):
+    """ """
+    for pool_type in rbd_obj.get("pool_types"):
+        rbd_config = kw.get("config", {}).get(pool_type, {})
+        multi_pool_config = getdict(rbd_config)
+        for pool, pool_config in multi_pool_config.items():
+            multi_image_config = getdict(pool_config)
+            for image in multi_image_config.keys():
+                log.info(f"Creating snapshot and clone for image {pool}/{image}")
+                rbd = rbd_obj.get("rbd")
+                # Start by creating cli for rbd feature commands
+                feature_spec = {
+                    "image-spec": f"{pool}/{image}",
+                    "features": "deep-flatten",
+                }
+                out, err = rbd.feature.disable(**feature_spec)
+                if out or err:
+                    log.error(f"Feature deep-flatten was not disabled for {image}")
+                    return 1
+                info_spec = {"image-or-snap-spec": f"{pool}/{image}", "format": "json"}
+                out, err = rbd.info(**info_spec)
+                if err:
+                    log.info(f"Error while fetching info for image {image}")
+                    return 1
+                log.info(f"Image info: {out}")
+
+                out_json = json.loads(out)
+                if "deep-flatten" in out_json["features"]:
+                    log.error(f"Feature deep-flatten was not disabled for {image}")
+                    return 1
+
+                snap_spec = f"{pool}/{image}@snap_{image}"
+                clone_spec = f"{pool}/clone_{image}"
+                create_snap_and_clone(rbd, snap_spec, clone_spec)
+                # Create snapshot for the clone
+                clone_snap_spec = {
+                    "snap-spec": f"{pool}/clone_{image}@snap_clone_{image}"
+                }
+                out, err = rbd.snap.create(**clone_snap_spec)
+                if out or err and "100% complete" not in err:
+                    log.error(f"Snapshot creation failed for {clone_snap_spec}")
+                    return 1
+
+                flatten_config = {"image-spec": clone_spec}
+                out, err = rbd.flatten(**flatten_config)
+                if out or err and "100% complete" not in err:
+                    log.error(f"Flatten clone failed for {clone_spec}")
+                    return 1
+
+                snap_conf = {"snap-spec": snap_spec}
+                out, err = rbd.snap.unprotect(**snap_conf)
+                if out or err:
+                    err_msg = f"Snapshot unprotect failed as expected for {snap_spec} since deep-flatten"
+                    err_msg += f" is not enabled on the image {image} and clone contains snapshots"
```

Comment:
instead of this we can create multiple line string.

Reply:
yes for code readability purpose.

Reply:
This is also readable according to me and since we are not following any standard at the CEPHCI level, I would like to retain this as is until we arrive at such a standard, Please let me know if you have any different view on this.